lib/matplotlib/axes/_axes.py

```diff
@@ -6320,25 +6320,34 @@ def _normalize_input(inp, ename='input'):
 
             # adopted from adjust_x/ylim part of the bar method
             if orientation == 'horizontal':
-                xmin0 = max(_saved_bounds[0]*0.9, minimum)
                 xmax = self.dataLim.intervalx[1]
-                for m in n:
-                    if np.sum(m) > 0:  # make sure there are counts
-                        xmin = np.amin(m[m != 0])
-                        # filter out the 0 height bins
-                xmin = max(xmin*0.9, minimum) if not input_empty else minimum
-                xmin = min(xmin0, xmin)
+                # no data or all bins have 0 counts
+                if input_empty or np.sum(m) == 0:
```

Comment:
What is `m` here?  `m` is used as an iteration variable in the other code branch.